planner/core/logical_plan_builder.go

```diff
@@ -183,6 +183,7 @@ func (b *PlanBuilder) buildResultSetNode(ctx context.Context, node ast.ResultSet
 				col.TblName = x.AsName
 			}
 		}
+		b.ctx.GetSessionVars().PlannerSelectBlockAsName[p.SelectBlockOffset()] = p.Schema().Columns[0].TblName
```

Comment:
Is it okay to use `Column[0].TblName` here?

Reply:
It's result set. So there's only two case, single datasource, multiple datasource with `AS` clause.